source/common/http/conn_manager_impl.cc

```diff
@@ -1529,7 +1524,63 @@ void ConnectionManagerImpl::ActiveStream::sendLocalReply(
     state_.saw_connection_close_ = true;
   }
 
-  stream_info_.setResponseCodeDetails(details);
+  if (response_headers_.get() == nullptr) {
+    // If the response has not started at all, send the response through the filter chain.
```

Comment:
something for my own knowledge, if you send the response through the filter chain, do you have to explicitly call `allowIteration()` in case it was stopped?